presidio-analyzer/analyzer/entity_recognizer.py

```diff
@@ -86,3 +95,153 @@ def to_dict(self):
+        """
+
+        if context_list is None:
+            return 0
+
+        lemmatized_keywords = self.__context_to_keywords(context_text)
+        if lemmatized_keywords is None:
+            return 0
+
+        similarity = 0.0
+        for context_keyword in lemmatized_keywords:
+            if context_keyword in context_list:
+                self.logger.info("Found context keyword '%s'", context_keyword)
+                similarity = 1
+                break
+
+        return similarity
+
+    def __extract_context(self, nlp_artifacts, word, start):
+        """ Extracts words surronding another given word.
+            The text from which the context is extracted is given in the nlp
+            doc
+            :param nlp_artifacts: An abstraction layer which holds different
+                                  items which are result of a NLP pipeline
+                                  execution on a given text
+            :param word: The word to look for context around
+            :param start: The start index of the word in the original text
+        """
+
+        if not nlp_artifacts.tokens:
+            self.logger.info('Skipping context extraction due to '
+                             'lack of NLP artifacts')
+            # if there are no nlp artifacts, this is ok, we can
+            # extract context and we return a valid, yet empty
+            # context
+            return ''
+
+        # Get the already prepared words in the given text, in their
+        # LEMMATIZED version
+        lemmatized_keywords = nlp_artifacts.keywords
+
+        found = False
+        # we use the known start index of the original word to find the actual
+        # token at that index, we are not checking for equivilance since the
+        # token might be just a substring of that word (e.g. for phone number
+        # 555-124564 the first token might be just '555')
+        # Note: we are iterating over the original tokens (not the lemmatized)
+        tokens = nlp_artifacts.tokens
+        tokens_indices = nlp_artifacts.tokens_indices
+        for i in range(len(nlp_artifacts.tokens)):
+            if ((tokens_indices[i] == start) or
+                    (tokens_indices[i] < start <
+                     tokens_indices[i] + len(tokens[i]))):
+                # found the interesting token, the one that around it
+                # we take n words
+                token_of_intrest = tokens[i]
+                found = True
+                break
+
+        token_count = 0
+        if not found:
+            raise ValueError("Did not find word '" + word + "' "
+                             "in the list of tokens altough it "
+                             "is expected to be found")
+
+        # now, iterate over the lemmatized text to find the location of that
+        # partial token
+        found = False
+        for keyword in lemmatized_keywords:
+            if keyword == token_of_intrest:
```

Comment:
Don't we still compare lemmas to original tokens? One simple workaround is to save token_of_interest as lemmatized_tokens[i] and not tokens[i]

Reply:
I changed the logic. see the latest revision